pod/pod.go

```diff
@@ -1,15 +1,59 @@
 package main
 
 import (
+	"log"
 	"os"
 	"os/signal"
 	"syscall"
 )
 
+func reapChildProcesses() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGCHLD)
+
+	for {
+		// Block until SIGCHLD signal is received.
+		<-c
+		reapChildProcessesHelper()
+	}
+}
+
+func reapChildProcessesHelper() {
+	for {
+		// Pid -1 means to wait for any child process.
+		// With syscall.WNOHANG option set, function will
+		// not block and will exit immediatelly if no child
+		// process has exited.
+		_, err := syscall.Wait4(-1, nil, syscall.WNOHANG, nil)
+
+		switch err {
+		case nil:
```

Comment:
We need to break out here when pid is 0 else we unnecessarily loop on the wait call. 
``` if pid == 0 {
          return
     }
```

Reply:
Done